docs/src/references/c8y-configuration-management.md

```diff
@@ -163,3 +279,232 @@ Note that:
+when transferred between the cloud and the child devices.
+* For these copies on the fly nothing is expected for the child-device configuration software.
+* A first special case is added to have the `c8y_configuration_plugin` requesting itself
+  a snapshot of the configuration for a child device if missing
+  (i.e. there is no `c8y-configuration-plugin.toml` for that device)
+* A second special case is also required to let the child device be able to send update
+  of its `$CHILD_DEVICE_ID/c8y-configuration-plugin.toml` on creation and updates.
+
+1. On reception of a `successful` status message
+   related to the `"type": "c8y-configuration-plugin"`,
+   received either for download on `tedge/$CHILD_DEVICE_ID/commands/req/config_update`
+   or for upload on `tedge/$CHILD_DEVICE_ID/commands/req/config_snapshot`.
+   1. The `c8y_configuration_plugin` takes a copy of the transferred file
+      that is under `$TEDGE_HTTP_ROOT/$CHILD_DEVICE_ID/config_update/c8y-configuration-plugin`
+      or `$TEDGE_HTTP_ROOT/$CHILD_DEVICE_ID/config_snapshop/c8y-configuration-plugin`
+      (depending on the file has been pushed by the cloud or the child device).
+   1. The `c8y_configuration_plugin` puts this copy under
+      `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID/c8y-configuration-plugin.toml`,
+      and starts to used it for an updated list of configuration files
+      (to which this file `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID/c8y-configuration-plugin.toml`
+      is implicitly added with type `c8y-configuration-plugin`).
+   1. If the request were triggered by the cloud,
+      the `c8y_configuration_plugin` proceeds on processing the `successful` status message,
+      notifying the cloud and removing the file `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID/c8y-configuration-plugin.toml`.
+      Note that the request might have been triggered by the main device on start,
+      or by the child device itself (see below),
+      in that case the cloud most not be notified. 
+1. On start, the `c8y_configuration_plugin` check for each child device `$CHILD_DEVICE_ID`,
+   listed in the directory `$TEDGE_CONFIG_DIR/c8y/`,
+   if there is a known `$CHILD_DEVICE_ID/c8y-configuration-plugin.toml`.
+   1. If the configuration is missing for some `$CHILD_DEVICE_ID`
+      (i.e. if the directory `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID` exists
+      but not the file `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID/c8y-configuration-plugin.toml`).
+   1. Then the `c8y_configuration_plugin` request this file from the child device
+      by sending an MQTT message on the topic `tedge/$CHILD_DEVICE_ID/commands/req/config_snapshot`.
+   1. The payload of this message is a JSON record with 2 required fields:
+       * `"url": "http://$TEDGE_HTTP/tedge/$CHILD_DEVICE_ID/config_snapshot/c8y-configuration-plugin"`
+       * `"type": "c8y-configuration-plugin"`    
+1. The child-device configuration software has nothing specific to do on requests
```

Comment:
I don't understand why you've written that statement that the child device "has nothing specific to do on requests for `"type": "c8y-configuration-plugin"`. They have to respond, right? For example, when the `config_snasphot` for this type is requested, the child device is expected to upload a TOML file in the prescribed format with the config list in it, right? Similarly, on a `config_update` request, the child device is supposed to update its own internal config list in whatever format that they maintain it in (may or may not be our prescribed TOML format)?

Reply:
What I meant is that the child-device behaves exactly as for any other snapshot request.